code/parse/sexp.h

```diff
@@ -979,28 +979,36 @@ const char *CTEXT(int n);
 #define CADDDDR(n)	CAR(CDDDDR(n))
 #define CADDDDDR(n)	CAR(CDDDDDR(n))
 
-#define REF_TYPE_SHIP		1
-#define REF_TYPE_WING		2
-#define REF_TYPE_PLAYER		3
-#define REF_TYPE_WAYPOINT	4
-#define REF_TYPE_PATH		5	// waypoint path
-
-#define SRC_SHIP_ARRIVAL	0x10000
-#define SRC_SHIP_DEPARTURE	0x20000
-#define SRC_WING_ARRIVAL	0x30000
-#define SRC_WING_DEPARTURE	0x40000
-#define SRC_EVENT				0x50000
-#define SRC_MISSION_GOAL	0x60000
-#define SRC_SHIP_ORDER		0x70000
-#define SRC_WING_ORDER		0x80000
-#define SRC_DEBRIEFING		0x90000
-#define SRC_BRIEFING			0xa0000
-#define SRC_UNKNOWN			0xffff0000
-#define SRC_MASK				0xffff0000
-#define SRC_DATA_MASK		0xffff
-
-#define SEXP_MODE_GENERAL	0
-#define SEXP_MODE_CAMPAIGN	1
+enum class SEXP_REF_TYPE
+{
+	SHIP = 1,
+	WING,
+	PLAYER,
+	WAYPOINT,
+	WAYPOINT_PATH
+};
+
+enum class SEXP_SRC
+{
+	NONE = 0,
+	SHIP_ARRIVAL,
+	SHIP_DEPARTURE,
+	WING_ARRIVAL,
+	WING_DEPARTURE,
+	EVENT,
+	MISSION_GOAL,
+	SHIP_ORDER,
+	WING_ORDER,
+	DEBRIEFING,
+	BRIEFING,
+	UNKNOWN
+};
+
+enum class SEXP_MODE
+{
+	GENERAL = 0,
+	CAMPAIGN
```

Comment:
Almost all scoped enums in FSO, at least as far as I am aware, have a snake_case class name and SCREAMING_SNAKE_CASE value names, so I'd prefer if the enum class names here were made lowercase

Reply:
As discussed, the name cases are a bit more mixed, but the ones in sexp.h at least are snake_case.  I've made the change.